modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -2231,6 +2232,27 @@ public ScheduledExecutorService streamerFlushExecutor() {
         return streamerFlushExecutor.get();
     }
 
+    @Override
+    public CompletableFuture<Long> estimatedSize() {
+        var invokeFutures = new CompletableFuture<?>[partitions];
+
+        for (int partId = 0; partId < partitions; partId++) {
```

Comment:
What if some partitions are unavailable? Is it intended to await all of them or we should also have best effort solution?

Reply:
How do we usually handle this? What will happen to the `SELECT *` query in this case? I think we should mimic the full scan behaviour here

Reply:
Implicit `Select *` is an RO wrapped operation, thus it might be executed on backup even without majority - in that case, behavior will differ. All in all that sounds like a follow up ticket that will add an ability to retrieve size from backups.

Reply:
Discussed this in person: there's no need for a best-effort approach, we will wait for the partition to appear